kommunicate/src/main/java/io/kommunicate/Kommunicate.java

```diff
@@ -158,18 +161,164 @@ public void onFailure(RegistrationResponse registrationResponse, Exception excep
+        } catch (ClassNotFoundException e) {
+            throw new KmException(e.getMessage());
+        }
+    }
+
     public static void loginAsVisitor(Context context, KMLoginHandler handler) {
         login(context, getVisitor(), handler);
     }
 
+    public static void launchConversationWithPreChat(final Context context, final ProgressDialog progressDialog, final KmCallback callback) {
+        final KMUser kmUser = getVisitor();
+        if (isLoggedIn(context)) {
+            String loggedInUserId = MobiComUserPreference.getInstance(context).getUserId();
+            if (loggedInUserId.equals(kmUser.getUserId())) {
+                launchChatDirectly(context, callback);
+            } else {
+                logout(context, new KMLogoutHandler() {
+                    @Override
+                    public void onSuccess(Context context) {
+                        loginUserWithKmCallBack(context, kmUser, callback);
+                    }
+
+                    @Override
+                    public void onFailure(Exception exception) {
+                        callback.onFailure(exception);
+                    }
+                });
+            }
+        } else {
+            checkForLeadCollection(context, progressDialog, kmUser, callback);
+        }
+
+    }
+
+    public static void checkForLeadCollection(final Context context, final ProgressDialog progressDialog, final KMUser kmUser, final KmCallback callback) {
+        new KmAppSettingTask(context, Applozic.getInstance(context).getApplicationKey(), new KmCallback() {
+            @Override
+            public void onSuccess(Object message) {
+                final KmAppSettingModel appSettingModel = (KmAppSettingModel) message;
+                if (appSettingModel != null && appSettingModel.getResponse() != null && appSettingModel.getChatWidget() != null) {
+                    if (appSettingModel.getResponse().isCollectLead()) {
+                        Utils.printLog(context, TAG, "Lead Collection is enabled..Launching Lead Collection");
+                        loginLeadUserAndOpenChat(context, appSettingModel, progressDialog, callback);
+                    } else {
+                        Utils.printLog(context, TAG, "Lead Collection is Disabled..Launching Random Login");
+                        loginUserWithKmCallBack(context, kmUser, callback);
+                    }
+                } else {
+                    Utils.printLog(context, TAG, "Failed to fetch App setting..Launching Random Login");
+                    loginUserWithKmCallBack(context, kmUser, callback);
+                }
+            }
+
+            @Override
+            public void onFailure(Object error) {
+                Utils.printLog(context, TAG, "Failed to fetch AppSetting");
+                loginUserWithKmCallBack(context, kmUser, callback);
+
+            }
+        }).execute();
+    }
+
+    // To Login the user from Lead Collection Details and Launch the Chat directly
```

Comment:
Can you also add method docs which includes argument details?

Reply:
@mukeshthawani  done